Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -5397,12 +5670,12 @@ protected override void UpdateTractiveForce(float elapsedClockSeconds, float t,
             }
 
             ApplyDirectionToTractiveForce();
-            
+
             LocomotiveAxles[0].DriveForceN = TractiveForceN;
         }
 
 
-        public override void AdvancedAdhesion(float elapsedClockSeconds)
+        public override void AdvancedAdhesion(float elapsedClockSeconds, int numberofengine)
```

Comment:
I'll see how can we refactor this method, current implementation (even before your changes) is not ideal. We should reuse as much as possible the code in AdvancedAdhesion(). Also I don't think adding numberofengine is a good idea, since for electric and diesel locomotives multiple engines already work without this parameter.